### PR TITLE
refactor: notify result vault

### PR DESCRIPTION
**Pull request by trkhoi**

### CODE REVIEW

File: pkg/entities/vault.go [hunk 1]
```diff
@@ -86,35 +86,66 @@ func (e *Entity) AddTreasurerToVault(req *request.AddTreasurerToVaultRequest) (*
...
+
+func (e *Entity) CreateTreasurerResult(req *request.CreateTreasurerResultRequest) error {
 	vault, err := e.repo.Vault.GetById(req.VaultId)
 	if err != nil {
 		e.log.Fields(logger.Fields{"req": req}).Errorf(err, "[entity.AddTreasurerToVault] - e.repo.Vault.GetById failed")
-		return nil, err
+		return err
+	}
+
+	action := "added"
```
Comment (anhnh12) [inline pkg/entities/vault.go:99]: constant for this?

File: pkg/entities/vault.go [hunk 2]
```diff
@@ -86,35 +86,66 @@ func (e *Entity) AddTreasurerToVault(req *request.AddTreasurerToVaultRequest) (*
...
+	thumbnail := "https://cdn.discordapp.com/attachments/1090195482506174474/1092703907911847976/image.png"
+	if req.Type == "remove" {
```
Comment (anhnh12) [inline pkg/entities/vault.go:101]: same

File: pkg/entities/vault.go [hunk 3]
```diff
@@ -86,35 +86,66 @@ func (e *Entity) AddTreasurerToVault(req *request.AddTreasurerToVaultRequest) (*
...
+		action = "removed"
```
Comment (anhnh12) [inline pkg/entities/vault.go:102]: same

File: pkg/entities/vault.go [hunk 4]
```diff
@@ -86,35 +86,66 @@ func (e *Entity) AddTreasurerToVault(req *request.AddTreasurerToVaultRequest) (*
...
+					Title:       fmt.Sprintf("<:revoke:967285238055174195> Treasurer was not %s", action),
+					Description: fmt.Sprintf("<@%s> has not been %s to **%s vault**", req.UserDiscordID, action, vault.Name),
+					Color:       0xFCD3C1,
+					Thumbnail: &discordgo.MessageEmbedThumbnail{
+						URL: thumbnail,
+					},
+					Timestamp: time.Now().Format("2006-01-02T15:04:05Z"),
+					Footer: &discordgo.MessageEmbedFooter{
+						Text: "Type /feedback to report",
+					},
```
[the remark this answers is not in the document]
Comment (trkhoi) [reply pkg/entities/vault.go:137]: just different msg but use same function sendDM so i think no need to seperate it

**Review by phucledien — APPROVED** [review]

File: pkg/entities/vault.go [hunk 5]
```diff
@@ -86,35 +86,66 @@ func (e *Entity) AddTreasurerToVault(req *request.AddTreasurerToVaultRequest) (*
...
 				},
 			},
-		},
-	})
+		}
+	}
+
+	err = e.svc.Discord.SendMessage(req.ChannelId, msg)
```
Comment (anhnh12) [inline pkg/entities/vault.go:143]: should be a function

**Review by anhnh12 — APPROVED** [review]

**Pull request merged**